mm2src/coins/utxo/rpc_clients/electrum_rpc/connection_managers/selective.rs

```diff
@@ -112,23 +112,28 @@ impl ConnectionManagerTrait for Arc<ConnectionManagerSelective> {
     fn weak_client(&self) -> &RwLock<Option<Weak<ElectrumClientImpl>>> { &self.electrum_client }
 
     async fn get_active_connections(&self) -> Vec<Arc<ElectrumConnection>> {
-        // We might have some other connected servers (e.g. was connected because of connection by address),
-        // but we will only return the one we know it's active right now. This also avoids sending pings to
-        // connections that are not used thus keeping their connections alive.
-        let maybe_active_connection = self
-            .active_address
-            .lock()
-            .unwrap()
-            .as_ref()
-            .and_then(|address| self.get_connection(address));
-        if let Some(active_connection) = maybe_active_connection {
-            if active_connection.is_connected().await {
-                // There is only one active connection at a time.
-                return vec![active_connection];
+        let trials = 10;
+        let sleep_time = 0.5;
+        for _ in 0..trials {
+            // We might have some other connected servers (e.g. was connected because of connection by address),
+            // but we will only return the one we know it's active right now. This also avoids sending pings to
+            // connections that are not used thus avoids keeping their connections alive.
+            let maybe_active_connection = self
+                .active_address
+                .lock()
+                .unwrap()
+                .as_ref()
+                .and_then(|address| self.get_connection(address));
+            if let Some(active_connection) = maybe_active_connection {
+                if active_connection.is_connected().await {
+                    // There is only one active connection at a time.
+                    return vec![active_connection];
+                }
             }
+            // We don't currently have an active connection.
+            self.trigger_reconnection();
+            Timer::sleep(sleep_time).await;
         }
```

Comment:
We can also make this forever-waiting, i think that would be optimal taking swap failures in mind. tests will probably need to be adjusted though since a faulty electrum will hang them if no timeout is used.